File: src/cpp/FindContinuousSequence.cpp
class Solution {
public:
    vector<vector<int> > FindContinuousSequence(int sum) {
      if (sum < 3) {
        return vector<vector<int> >();
      }

      int small = 1;
      int big = 2;
      int mid = (1 + sum) / 2;
      int currSum = small + big;

      vector<vector<int> > result;

      while (small < mid) {
        if (currSum == sum) {
          vector<int> sequence;
          for (int i = small; i <= big; ++i) {
            sequence.push_back(i);
          }
          result.push_back(sequence);
        }

        while (currSum > sum && small < mid) {
          currSum -= small;
          small++;

          if (currSum == sum) {
            vector<int> sequence;
            for (int i = small; i <= big; ++i) {
              sequence.push_back(i);
            }
            result.push_back(sequence);
          }
        }

        big++;
        currSum += big;
      }

      return result;

    }
};

File: src/java/FindContinuousSequence.java
import java.util.ArrayList;
public class Solution {
    public ArrayList<ArrayList<Integer> > FindContinuousSequence(int sum) {
        ArrayList<ArrayList<Integer>> result = new ArrayList<>();

        if (sum < 3) {
            return result;
        }

        int small = 1;
        int big = 2;
        int mid = (sum + 1) / 2;
        int currSum = small + big;

        while (small < mid) {
            while (small < mid && currSum > sum) {
                currSum -= small;
                small++;
            }

            if (currSum == sum) {
                addToList(result, small, big);
            }

            big++;
            currSum += big;
        }

        return result;
    }

    private void addToList(ArrayList<ArrayList<Integer>> result, int small, int big) {
        ArrayList<Integer> list = new ArrayList<>();
        for (int i = small; i <= big; i++) {
            list.add(i);
        }
        result.add(list);
    }
}
